dmutils/filters.py

```diff
@@ -24,18 +25,19 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '{0}<a href="{1}" rel="external">{2}</a>'
         formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+        textArray = url_match.split(text)
```

Comment:
(nitpicking) Could we stick with underscore variable names for `textArray` and `formattedText`?